### PR TITLE
Fixed troubleshooting header id and added pre-commit installation troubleshooting

### PR DESCRIPTION
**Pull request by tbhaxor**
## Description

This pull request contains the two this

1. Fix for the id of the "LightDM" troubleshooting. `#id1`
2. Add instructions to fix the `pre-commit` installation [here](https://github.com/Bash-it/bash-it/issues/1696#issuecomment-1035968473)  

## How Has This Been Tested?

Locally on Linux using docs build command

## Screenshots (if appropriate):

image

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] If I have added a new file, I also added it to ``clean_files.txt`` and formatted it using ``lint_clean_files.sh``.
- [ ] I have added tests to cover my changes, and all the new and existing tests pass.

### CODE REVIEW

**Review by cornfeedhobo — APPROVED** [review]
@davidpfarrell LGTM, but could you please give this a look over?

File: docs/troubleshooting.rst [hunk 1]
```diff
@@ -6,11 +6,19 @@ Troubleshooting Guide
...
 
 I'm stuck in the LightDM login screen after setting up bash-it
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 **Possible issue**\ : `#672 <https://github.com/Bash-it/bash-it/issues/672>`_
 
 **Solution**\ : Check `this comment <https://github.com/Bash-it/bash-it/issues/672#issuecomment-257870653>`_ for detailed information about the cause and solution for this issue.
+
+Pre-commit installation is failed with "unrecognized option --gdwarf-5" error
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (davidpfarrell) [inline docs/troubleshooting.rst:20]: /nitpick Looks like about 5 too many trailing `^`

File: docs/troubleshooting.rst [hunk 2]
```diff
@@ -6,11 +6,19 @@ Troubleshooting Guide
 Table of Contents
 -----------------
 
-* `I'm stuck in the LightDM login screen after setting up bash-it. <im-stuck-in-the-lightdm-login-screen-after-setting-up-bash-it>`_
+* `I'm stuck in the LightDM login screen after setting up bash-it. <#id1>`_
+* `Pre-commit installation is failed with "unrecognized option --gdwarf-5" error. <#id3>`_
```
Comment (davidpfarrell) [inline docs/troubleshooting.rst:10]: Why is it `#id3` instead of `#id2` ?

**Review by davidpfarrell — CHANGES_REQUESTED** [review]
Greetings !

I'm looking at the [reStructuredText Markup Specification
](https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html) page and can't find reference to this `<#..>` usage - Can you point me to it? And/or maybe other online examples of this usage?

File: docs/troubleshooting.rst (hunk 2 above)
Comment (tbhaxor) [reply docs/troubleshooting.rst:10]: This is called id attribute scroll/focus. I checked that in restructured text, it is setting the id attribute to `#id1`. Which is wrong in the current https://bash-it.readthedocs.io/en/latest/troubleshooting/im-stuck-in-the-lightdm-login-screen-after-setting-up-bash-it redirected to 404 whereas https://bash-it.readthedocs.io/en/latest/troubleshooting/#id1 takes to the correct heading on the troubleshooting page.

**Comment by tbhaxor** [comment]
> Greetings !
> 
> I'm looking at the [reStructuredText Markup Specification ](https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html) page and can't find reference to this `<#..>` usage - Can you point me to it? And/or maybe other online examples of this usage?

`Content <link>`_ is basically a syntax in rst which gets converted to `[Content](link)` in markdown or `<a href="link">Content</a>` in html. You can have any type of link
- Absolute &rarr; https://example.com/
- Relative &rarr; /my-page
- Hash &rarr; #element-id

File: docs/troubleshooting.rst (hunk 2 above)
Comment (tbhaxor) [reply docs/troubleshooting.rst:10]: These ids' are of the section headers generated by the rst. Even I am wondering why it didn't slugify the header text instead.

File: docs/troubleshooting.rst (hunk 1 above)
Comment (tbhaxor) [reply docs/troubleshooting.rst]: @davidpfarrell fixed

**Comment by tbhaxor** [comment]
Timeout

**Pull request closed without merging**